### DISCUSSION TITLE
Changes to extending ApplicationShell in 1.38

### DISCUSSION THREAD

**Discussion by BarryDrez** [eclipse-theia/theia#12639] [General] [answered]
I have been extending the ApplicationShell in my custom extension using the following in my frontend module:
```
    bind(JavaServiceApplicationShell).toSelf().inSingletonScope();
    rebind(ApplicationShell).to(JavaServiceApplicationShell).inSingletonScope();
```
In 1.38, it seems that my code is being ignored.  The ApplicationShell is being extended by ApplicationShellWithToolbarOverride.

Can someone please comment on how can I overcome this?

**Comment by msujew** [comment, marked as answer]
Did you recently add the `@theia/toolbar` package to your application? We have had that package for a long time, and nothing changed in regards to that in Theia 1.38.0.

But yes, the package overrides the application shell with a different one.

**Comment by BarryDrez** [comment]
@msujew Yes, you are right.  I added `@theia/toolbar`.  I copied the dependencies from Theia Blueprint (`theia-blueprint/applications/browser/package.json`).  After removing `@theia/toolbar` from my package.json, it works.  Thanks very much!
